packages/builder/builder-webpack-provider/src/plugins/sass.ts

```diff
@@ -1,25 +1,20 @@
-import path from 'path';
-import { pathToFileURL } from 'url';
 import {
   isUseCssSourceMap,
   SASS_REGEX,
   getSassLoaderOptions,
 } from '@modern-js/builder-shared';
 import type { BuilderPlugin } from '../types';
 
-/** fix issue about dart2js: https://github.com/dart-lang/sdk/issues/27979 */
-export function patchGlobalLocation() {
-  const href = `${pathToFileURL(process.cwd()).href}${path.sep}`;
-  // @ts-expect-error
-  global.location ||= {};
-  global.location.href ||= href;
-}
-
 export function builderPluginSass(): BuilderPlugin {
   return {
     name: 'builder-plugin-sass',
-    setup(api) {
+    async setup(api) {
+      const { patchGlobalLocation, unpatchGlobalLocation } = await import(
+        '@modern-js/utils'
+      );
       patchGlobalLocation();
+      api.onAfterBuild(unpatchGlobalLocation);
```

Comment:
If the code changes and recompilation, does the location no longer exist at this time, can sass work ?

Reply:
I think it won't be executed in watch mode:

https://github.com/web-infra-dev/modern.js/blob/48b036e034743ba5dee6f1b5192a15c8e375cefe/packages/builder/builder-webpack-provider/src/core/build.ts#L81-L92

Reply:
If this code will not be executed in watch mode, does that mean that it will not be executed in the dev? 

It seem not to be able to fix the current problem.

Reply:
Now use webpack hooks instead:

https://github.com/web-infra-dev/modern.js/blob/8ebbeb499342a8fde0edc6ad649513eea03e6bcc/packages/builder/builder-shared/src/patch.ts#L36-L48